avro/tests/to_from_avro_datum_schemata.rs

```diff
@@ -110,6 +110,7 @@ fn test_avro_3683_multiple_schemata_writer_reader() -> TestResult {
     let mut writer = Writer::with_schemata(schema_b, schemata.clone(), &mut output, Codec::Null);
     writer.append(record.clone())?;
     writer.flush()?;
+    let _ = writer.into_inner()?;
```

Comment:
What kind of magic happens here ?
Why calling `into_inner()` makes it possible to use `output` again ?

Reply:
> now has a Drop implementation, it lives until the end of the function

Could you please point me to the Rust documentation confirming this ?

Reply:
See this playground example with `BufWriter`: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=c8318f0b348bf8be78536bd33b619b3e
If you look at the compiler error, it explains the issue. I'm not sure where this would be documented.